lib/segment/src/data_types/vectors.rs

```diff
@@ -197,13 +197,25 @@ pub type MultiDenseVector = TypedMultiDenseVector<VectorElementType>;
 
 impl<T: PrimitiveVectorElement> TypedMultiDenseVector<T> {
     pub fn new(flattened_vectors: TypedDenseVector<T>, dim: usize) -> Self {
-        assert!(flattened_vectors.len() % dim == 0, "Invalid vector length");
+        assert_eq!(flattened_vectors.len() % dim, 0, "Invalid vector length");
         Self {
             inner_vector: flattened_vectors,
```

Comment:
Worth renaming the field itself to `flat_vector`?

Reply:
I like `flattened_vectors` because the convey the plural aspect of multivectors.
Why do you prefer `flat_vector`? :)

Reply:
Yeah, `flattened_vectors` too, but I meant renaming `inner_vector`